build-all.sh

```diff
@@ -67,26 +66,41 @@ fi
 
 exec >	>(tee -a "$BUILDALL_DIR"/ALL.out)
 exec 2> >(tee -a "$BUILDALL_DIR"/ALL.err >&2)
-trap 'echo ERROR: See $BUILDALL_DIR/${PKG}.err' ERR
+trap 'echo ERROR: See $BUILDALL_DIR/${PKG}.out' ERR
 
 while read -r PKG PKG_DIR; do
 	# Check build status (grepping is a bit crude, but it works)
-	if [ -e "$BUILDSTATUS_FILE" ] && grep "^$PKG\$" "$BUILDSTATUS_FILE" >/dev/null; then
+	if [ -e "$BUILDSTATUS_FILE" ] && grep -q "^$PKG\$" "$BUILDSTATUS_FILE"; then
 		echo "Skipping $PKG"
 		continue
 	fi
 
+	# Start building
+	if [ -n "${TERMUX_DEBUG_BUILD}" ]; then
+		echo "\"$BUILDSCRIPT\" -a \"$TERMUX_ARCH\" $TERMUX_DEBUG_BUILD ${TERMUX_OUTPUT_DIR+-o $TERMUX_OUTPUT_DIR} $TERMUX_INSTALL_DEPS \"$PKG_DIR\""
+	fi
 	echo -n "Building $PKG... "
 	BUILD_START=$(date "+%s")
-	bash -x "$BUILDSCRIPT" -a "$TERMUX_ARCH" $TERMUX_DEBUG_BUILD \
+	"$BUILDSCRIPT" -a "$TERMUX_ARCH" $TERMUX_DEBUG_BUILD \
 		${TERMUX_OUTPUT_DIR+-o $TERMUX_OUTPUT_DIR} $TERMUX_INSTALL_DEPS "$PKG_DIR" \
-		> "$BUILDALL_DIR"/"${PKG}".out 2> "$BUILDALL_DIR"/"${PKG}".err
+		&> "$BUILDALL_DIR"/"${PKG}".out
```

Comment:
Removing .err log is unrelated to cyclic dependencies, and should not be part of the same commit

Reply:
This is a separate change to `build-all.sh`. As practice has shown, it is not very convenient to look at the error log without understanding what happened before. In general, is it convenient for anyone to work with such log formats (I’m interested)?

Reply:
I agree with you that having different logs is not very convenient. It can be useful to get a quick overview of what the error was, maybe it was just wrong checksum or failed download, but for deeper issues and further debugging the context is needed. 

In any case, the change is not described in the commit message, and should not be part of the same commit. I (and you as well probably) often have to rebase branches or cherry-pick commits between branches. This is a lot easier if all commits are small and self-explanatory, as it is easier to resolve conflicts in that case. Smaller commits are also easier to review.

Reply:
I added information about all the changes in the commit of this pr. Is it clearer about the changes?

Reply:
The title "Script improvements and fixes" very briefly tells what's going on here. I made this title because in this PR there is not only a change in `buildorder.py`, but also in other places for a different reason.